tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,338 @@
+    ns = namespace["ns"]
+    keyvaluepair = {"testadelete": "valuedelete"}
+
+    secret = create_secret(keyvaluepair, singlenamespace=True)
+    delete_secret(p_client, secret, ns, keyvaluepair)
+
+
+def test_edit_secret_all_ns():
+
+    p_client = namespace["p_client"]
+    name = random_test_name("default")
+    keyvaluepair = {"testall": "valueall"}
+    cluster = namespace["cluster"]
+    project = namespace["project"]
+    c_client = namespace["c_client"]
+
+    # Create a namespace
+    new_ns = create_ns(c_client, cluster, project)
+
+    secret = create_secret(keyvaluepair)
+
+    newkeyvaluepair = {"testallnew": "valueallnew"}
+    updated_dict = {"testall": "valueall", "testallnew": "valueallnew"}
+    updated_secret = p_client.update(secret, name=name, namespaceId='NULL',
+                                     stringData=newkeyvaluepair)
+
+    assert updated_secret['baseType'] == "secret"
+    updatedsecretdata = updated_secret['data']
+
+    print updatedsecretdata
+
+    finalkeyvaluepair = {}
+    for k, v in updatedsecretdata.items():
+        keytest = k
+        vdecoded = base64.b64decode(v)
+        finalkeyvaluepair.update({keytest: vdecoded})
+
+    assert len(updatedsecretdata.items()) == 2
+    print finalkeyvaluepair
+    assert cmp(updated_dict, finalkeyvaluepair) == 0
+
+    # Create workloads using updated secret in the existing namespace
+    create_and_validate_workload_with_secret_as_volume(p_client, secret,
+                                                       new_ns,
+                                                       finalkeyvaluepair)
+
+    create_and_validate_workload_with_secret_as_env_variable(p_client, secret,
+                                                             new_ns,
+                                                             finalkeyvaluepair)
+
+    # Create a new namespace and workloads in the new namespace using secret
+    new_ns1 = create_ns(c_client, cluster, project)
+
+    create_and_validate_workload_with_secret_as_volume(p_client, secret,
+                                                       new_ns1,
+                                                       finalkeyvaluepair)
+    create_and_validate_workload_with_secret_as_env_variable(p_client, secret,
+                                                             new_ns1,
+                                                             finalkeyvaluepair)
+
+
+def test_edit_secret_single_ns():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("default")
+    keyvaluepair = {"testall": "valueall"}
+
+    secret = create_secret(keyvaluepair, singlenamespace=True)
+
+    newkeyvaluepair = {"testallnew": "valueallnew"}
+    updated_dict = {"testall": "valueall", "testallnew": "valueallnew"}
+    updated_secret = p_client.update(secret, name=name,
+                                     namespaceId=ns['name'],
+                                     stringData=newkeyvaluepair)
```

Comment:
@soumyalj Use data instead of stringData for updates similar to how UI workflow uses.